mpqp/core/instruction/gates/custom_gate.py

```diff
@@ -12,19 +17,49 @@
 
 @typechecked
 class CustomGate(Gate):
-    """Custom gates allow you to define your own gates.
+    """Custom gates allow you to define your own unitary gates.
 
     Args:
-        definition: The matrix (this is the only way supported to now) semantics of the gate.
+        definition: The UnitaryMatrix (only way supported for now) description of the gate.
         targets: The qubits on which the gate operates.
         label: The label of the gate. Defaults to None.
+
+    Raises:
+        ValueError: the target qubits must be contiguous and in order, and must match the size of the UnitaryMatrix
+
```

Comment:
Not mandatory, we just need to re-use the work you did to generate any matrix from a gate, playing with the targets etc.

However, I think this should be done on another branch, after merging this one, because the work may not be straightfoward, when dealing with qiskit qubit ordering and change in the size of the CustomGate.

I create a task in Trello.